### ISSUE TITLE
After upgrading to 3.5.1 from 3.3.x HA fails to show size and sysinfo - [Neo.DatabaseError.General.UnknownError]

### ISSUE THREAD

**Issue by gpierrick** [neo4j/neo4j#12121]
- Neo4j version: 3.5.1
- Operating system: Ubuntu 16.04

Followed this https://neo4j.com/docs/operations-manual/current/ha-cluster/upgrade/ to upgrade to latest version

- I started the master
- I started each slave with an empty database folder
- Cluster starts syncing and starts successfully **but** the following fails to show

![image](https://user-images.githubusercontent.com/1524537/50579310-ea523600-0e11-11e9-8d6f-3aad5036e8d4.png)

and sysinfo fails to show anything in the browser.

What's weird is the nodes are there and cypher queries are working.

**Things I did**

- I performed a few backups and ran consistency which worked fine
- I ended up restarting in SINGLE mode to bring our application online

Stack trace below from the debug logs

```
Exception, reference 5db6a870-e1c9-4fea-b47f-970cdcdc85f6.
2019-01-02 00:50:27.981+0000 ERROR [o.n.b.v.r.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: java.lang.NullPointerException, reference 5db6a870-e1c9-4fea-b47f-970cdcdc85f6. java.lang.NullPointerException
javax.management.RuntimeMBeanException: java.lang.NullPointerException
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.rethrow(DefaultMBeanServerInterceptor.java:839)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.rethrowMaybeMBeanException(DefaultMBeanServerInterceptor.java:852)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getAttribute(DefaultMBeanServerInterceptor.java:651)
        at com.sun.jmx.mbeanserver.JmxMBeanServer.getAttribute(JmxMBeanServer.java:678)
        at org.neo4j.kernel.builtinprocs.JmxQueryProcedure.toNeo4jValue(JmxQueryProcedure.java:135)
        at org.neo4j.kernel.builtinprocs.JmxQueryProcedure.toNeo4jValue(JmxQueryProcedure.java:123)
        at org.neo4j.kernel.builtinprocs.JmxQueryProcedure.lambda$apply$0(JmxQueryProcedure.java:95)
        at org.neo4j.collection.RawIterator$2.fetchNextOrNull(RawIterator.java:89)
        at org.neo4j.collection.PrefetchingRawIterator.peek(PrefetchingRawIterator.java:50)
        at org.neo4j.collection.PrefetchingRawIterator.hasNext(PrefetchingRawIterator.java:36)
        at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$$anon$1.hasNext(CallSupport.scala:167)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$$anon$1$$anonfun$hasNext$1.apply$mcZ$sp(ExceptionTranslationSupport.scala:49)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$$anon$1$$anonfun$hasNext$1.apply(ExceptionTranslationSupport.scala:49)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$$anon$1$$anonfun$hasNext$1.apply(ExceptionTranslationSupport.scala:49)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$class.translateException(ExceptionTranslationSupport.scala:33)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext.translateException(ExceptionTranslatingQueryContext.scala:41)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$$anon$1.hasNext(ExceptionTranslationSupport.scala:49)
        at org.neo4j.cypher.internal.runtime.DbmsCallMode.call(ProcedureCallMode.scala:106)
        at org.neo4j.cypher.internal.runtime.DbmsCallMode.callProcedure(ProcedureCallMode.scala:96)
        at org.neo4j.cypher.internal.compatibility.v3_5.runtime.executionplan.procs.ProcedureCallRuntimeResult.executeCall(ProcedureCallRuntimeResult.scala:70)
        at org.neo4j.cypher.internal.compatibility.v3_5.runtime.executionplan.procs.ProcedureCallRuntimeResult.<init>(ProcedureCallRuntimeResult.scala:64)
        at org.neo4j.cypher.internal.compatibility.v3_5.runtime.executionplan.procs.ProcedureCallExecutionPlan.run(ProcedureCallExecutionPlan.scala:71)
        at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery$$anonfun$execute$3.apply(CypherCurrentCompiler.scala:205)
        at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery$$anonfun$execute$3.apply(CypherCurrentCompiler.scala:190)
        at org.neo4j.cypher.exceptionHandler$runSafely$.apply(exceptionHandler.scala:89)
        at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery.execute(CypherCurrentCompiler.scala:224)
        at org.neo4j.cypher.internal.ExecutionEngine.execute(ExecutionEngine.scala:102)
        at org.neo4j.cypher.internal.javacompat.ExecutionEngine.executeQuery(ExecutionEngine.java:85)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachineV1SPI$BoltResultHandleV1.start(TransactionStateMachineV1SPI.java:189)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State.startExecution(TransactionStateMachine.java:506)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.execute(TransactionStateMachine.java:300)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.run(TransactionStateMachine.java:269)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine.run(TransactionStateMachine.java:110)
        at org.neo4j.bolt.v3.runtime.ReadyState.processRunMessage(ReadyState.java:85)
        at org.neo4j.bolt.v3.runtime.ReadyState.processUnsafe(ReadyState.java:56)
        at org.neo4j.bolt.v3.runtime.FailSafeBoltStateMachineState.process(FailSafeBoltStateMachineState.java:48)
        at org.neo4j.bolt.v1.runtime.BoltStateMachineV1.nextState(BoltStateMachineV1.java:144)
        at org.neo4j.bolt.v1.runtime.BoltStateMachineV1.process(BoltStateMachineV1.java:92)
        at org.neo4j.bolt.messaging.BoltRequestMessageReader.lambda$doRead$1(BoltRequestMessageReader.java:89)
        at org.neo4j.bolt.runtime.MetricsReportingBoltConnection.lambda$enqueue$0(MetricsReportingBoltConnection.java:68)
        at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:191)
        at org.neo4j.bolt.runtime.MetricsReportingBoltConnection.processNextBatch(MetricsReportingBoltConnection.java:86)
        at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:139)
        at org.neo4j.bolt.runtime.ExecutorBoltScheduler.executeBatch(ExecutorBoltScheduler.java:171)
        at org.neo4j.bolt.runtime.ExecutorBoltScheduler.lambda$scheduleBatchOrHandleError$2(ExecutorBoltScheduler.java:154)
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
        at org.neo4j.kernel.ha.management.BranchedStoreBean$BranchedStoreImpl.getBranchedStores(BranchedStoreBean.java:97)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.reflect.misc.Trampoline.invoke(MethodUtil.java:71)
        at sun.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.reflect.misc.MethodUtil.invoke(MethodUtil.java:275)
        at com.sun.jmx.mbeanserver.ConvertingMethod.invokeWithOpenReturn(ConvertingMethod.java:193)
        at com.sun.jmx.mbeanserver.ConvertingMethod.invokeWithOpenReturn(ConvertingMethod.java:175)
        at com.sun.jmx.mbeanserver.MXBeanIntrospector.invokeM2(MXBeanIntrospector.java:117)
        at com.sun.jmx.mbeanserver.MXBeanIntrospector.invokeM2(MXBeanIntrospector.java:54)
        at com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.java:237)
        at com.sun.jmx.mbeanserver.PerInterface.getAttribute(PerInterface.java:83)
        at com.sun.jmx.mbeanserver.MBeanSupport.getAttribute(MBeanSupport.java:206)
        at javax.management.StandardMBean.getAttribute(StandardMBean.java:372)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getAttribute(DefaultMBeanServerInterceptor.java:647)
        ... 46 more
2019-01-02 00:50:30.859+0000 ERROR [o.n.b.v.r.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: java.lang.NullPointerException, reference 89ab9af3-4e24-4812-82b6-f55ae877dc59.
2019-01-02 00:50:30.859+0000 ERROR [o.n.b.v.r.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: java.lang.NullPointerException, reference 89ab9af3-4e24-4812-82b6-f55ae877dc59. java.lang.NullPointerException
javax.management.RuntimeMBeanException: java.lang.NullPointerException
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.rethrow(DefaultMBeanServerInterceptor.java:839)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.rethrowMaybeMBeanException(DefaultMBeanServerInterceptor.java:852)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getAttribute(DefaultMBeanServerInterceptor.java:651)
        at com.sun.jmx.mbeanserver.JmxMBeanServer.getAttribute(JmxMBeanServer.java:678)
        at org.neo4j.kernel.builtinprocs.JmxQueryProcedure.toNeo4jValue(JmxQueryProcedure.java:135)
        at org.neo4j.kernel.builtinprocs.JmxQueryProcedure.toNeo4jValue(JmxQueryProcedure.java:123)
        at org.neo4j.kernel.builtinprocs.JmxQueryProcedure.lambda$apply$0(JmxQueryProcedure.java:95)
        at org.neo4j.collection.RawIterator$2.fetchNextOrNull(RawIterator.java:89)
        at org.neo4j.collection.PrefetchingRawIterator.peek(PrefetchingRawIterator.java:50)
        at org.neo4j.collection.PrefetchingRawIterator.hasNext(PrefetchingRawIterator.java:36)
        at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$$anon$1.hasNext(CallSupport.scala:167)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$$anon$1$$anonfun$hasNext$1.apply$mcZ$sp(ExceptionTranslationSupport.scala:49)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$$anon$1$$anonfun$hasNext$1.apply(ExceptionTranslationSupport.scala:49)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$$anon$1$$anonfun$hasNext$1.apply(ExceptionTranslationSupport.scala:49)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$class.translateException(ExceptionTranslationSupport.scala:33)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext.translateException(ExceptionTranslatingQueryContext.scala:41)
        at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$$anon$1.hasNext(ExceptionTranslationSupport.scala:49)
        at org.neo4j.cypher.internal.runtime.DbmsCallMode.call(ProcedureCallMode.scala:106)
        at org.neo4j.cypher.internal.runtime.DbmsCallMode.callProcedure(ProcedureCallMode.scala:96)
        at org.neo4j.cypher.internal.compatibility.v3_5.runtime.executionplan.procs.ProcedureCallRuntimeResult.executeCall(ProcedureCallRuntimeResult.scala:70)
        at org.neo4j.cypher.internal.compatibility.v3_5.runtime.executionplan.procs.ProcedureCallRuntimeResult.<init>(ProcedureCallRuntimeResult.scala:64)
        at org.neo4j.cypher.internal.compatibility.v3_5.runtime.executionplan.procs.ProcedureCallExecutionPlan.run(ProcedureCallExecutionPlan.scala:71)
        at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery$$anonfun$execute$3.apply(CypherCurrentCompiler.scala:205)
        at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery$$anonfun$execute$3.apply(CypherCurrentCompiler.scala:190)
        at org.neo4j.cypher.exceptionHandler$runSafely$.apply(exceptionHandler.scala:89)
        at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery.execute(CypherCurrentCompiler.scala:224)
        at org.neo4j.cypher.internal.ExecutionEngine.execute(ExecutionEngine.scala:102)
        at org.neo4j.cypher.internal.javacompat.ExecutionEngine.executeQuery(ExecutionEngine.java:85)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachineV1SPI$BoltResultHandleV1.start(TransactionStateMachineV1SPI.java:189)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State.startExecution(TransactionStateMachine.java:506)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.execute(TransactionStateMachine.java:300)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.run(TransactionStateMachine.java:269)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine.run(TransactionStateMachine.java:110)
        at org.neo4j.bolt.v3.runtime.ReadyState.processRunMessage(ReadyState.java:85)
        at org.neo4j.bolt.v3.runtime.ReadyState.processUnsafe(ReadyState.java:56)
        at org.neo4j.bolt.v3.runtime.FailSafeBoltStateMachineState.process(FailSafeBoltStateMachineState.java:48)
        at org.neo4j.bolt.v1.runtime.BoltStateMachineV1.nextState(BoltStateMachineV1.java:144)
        at org.neo4j.bolt.v1.runtime.BoltStateMachineV1.process(BoltStateMachineV1.java:92)
        at org.neo4j.bolt.messaging.BoltRequestMessageReader.lambda$doRead$1(BoltRequestMessageReader.java:89)
        at org.neo4j.bolt.runtime.MetricsReportingBoltConnection.lambda$enqueue$0(MetricsReportingBoltConnection.java:68)
        at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:191)
        at org.neo4j.bolt.runtime.MetricsReportingBoltConnection.processNextBatch(MetricsReportingBoltConnection.java:86)
        at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:139)
        at org.neo4j.bolt.runtime.ExecutorBoltScheduler.executeBatch(ExecutorBoltScheduler.java:171)
        at org.neo4j.bolt.runtime.ExecutorBoltScheduler.lambda$scheduleBatchOrHandleError$2(ExecutorBoltScheduler.java:154)
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
        at org.neo4j.kernel.ha.management.BranchedStoreBean$BranchedStoreImpl.getBranchedStores(BranchedStoreBean.java:97)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.reflect.misc.Trampoline.invoke(MethodUtil.java:71)
        at sun.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.reflect.misc.MethodUtil.invoke(MethodUtil.java:275)
        at com.sun.jmx.mbeanserver.ConvertingMethod.invokeWithOpenReturn(ConvertingMethod.java:193)
        at com.sun.jmx.mbeanserver.ConvertingMethod.invokeWithOpenReturn(ConvertingMethod.java:175)
        at com.sun.jmx.mbeanserver.MXBeanIntrospector.invokeM2(MXBeanIntrospector.java:117)
        at com.sun.jmx.mbeanserver.MXBeanIntrospector.invokeM2(MXBeanIntrospector.java:54)
        at com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.java:237)
        at com.sun.jmx.mbeanserver.PerInterface.getAttribute(PerInterface.java:83)
        at com.sun.jmx.mbeanserver.MBeanSupport.getAttribute(MBeanSupport.java:206)
        at javax.management.StandardMBean.getAttribute(StandardMBean.java:372)
        at com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getAttribute(DefaultMBeanServerInterceptor.java:647)
        ... 46 more
```

**Should I be worried?**

**Comment by rheusinkveld** [comment]
Hi @gpierrick. I have exactly the same issue after upgrading to 3.5.1 and also have the exact same stack traces in the log file. All cypher queries seem to be working fine. I'll create a support case with Neo4j and post back if I hear from them.

**Comment by rheusinkveld** [comment]
Neo4j support responded to my case and it seems to be happening for all HA environments. It's reported to engineering requesting a fix.

**Comment by gpierrick** [comment]
Hi @rheusinkveld thanks for your response and submitting a ticket! I guess I'll be waiting for a fix and run Neo4j in single for now

**Comment by gpierrick** [comment]
Is the preferred way to set up a cluster now causal?

**Comment by gpierrick** [comment]
Any updates on this? @rheusinkveld

**Comment by rheusinkveld** [comment]
Latest I got from support is that a fix will probably be included in the next release (3.5.2).
They also let me know that the HA cluster setup is deprecated and will be removed in Neo4j 4.0. See also https://neo4j.com/docs/operations-manual/current/ha-cluster/

**Comment by gpierrick** [comment]
Thanks @rheusinkveld . Did they give you any timelines? Are you currently HA regardless of this bug?

**Comment by rheusinkveld** [comment]
@gpierrick They didn't give me a specific date. The indication they gave me was late January to mid February. For us it only impacts our test environment, so we still have that running in HA setup. 
We probably look into migrating to a causal cluster instead of waiting for the fix.

**Comment by gpierrick** [comment]
Thanks @rheusinkveld

**Comment by gpierrick** [comment]
Was this fixed in 3.5.2?

**Comment by rheusinkveld** [comment]
Hi @gpierrick. I've upgraded to 3.5.2 and can confirm that this was fixed.

**Comment by gpierrick** [comment]
Thanks for confirming @rheusinkveld !

**Comment by ariffewear** [comment]
Hi guys, I'm new to neo4j and have install 3.5.3 enterprise with HA enable but still facing the exact problem. It is save to use HA in this kind of situation?

**Comment by rheusinkveld** [comment]
Hi @ariffewear. I know it was fixed in 3.5.2, but I'm not sure if it was broken again in 3.5.3.
We switched away from HA to causal clustering. If you are just starting with a new cluster I would suggest to do the same. It's the recommended way nowadays: 
https://neo4j.com/docs/operations-manual/current/clustering/

If that's not an option I would suggest to create a new issue or to raise the issue here: https://community.neo4j.com
